chain/fil/address/address.go

```diff
@@ -0,0 +1,365 @@
+package address
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"math/bits"
+	"strconv"
+	"strings"
+
+	xc "github.com/cordialsys/crosschain"
+	xcaddress "github.com/cordialsys/crosschain/address"
+	"github.com/dchest/blake2b"
+	secp "github.com/decred/dcrd/dcrec/secp256k1/v4"
+)
+
+const (
+	// An ID address
+	ProtocolId = 0
+	// A wallet address generated from a secp256k public key
+	ProtocolSecp256k1 = 1
+	// An actor address.
+	ProtocolActor = 2
+	// A wallet address generated from BLS public key.
+	ProtocolBls = 3
+	// A delegated address for user-defined foreign actors:
+	// For example f410 is and Ethereum compatibile address space
+	ProtocolDelegated    = 4
+	EncodeStd            = "abcdefghijklmnopqrstuvwxyz234567"
+	BlsPublicKeyBytesLen = 48
+	// MaxSubaddressLen is the maximum length of a delegated address's sub-address.
+	MaxSubaddressLen = 54
+	// ChecksumHashLength defines the hash length used for calculating address checksums.
+	ChecksumHashLength = 4
+)
+
+var payloadHashConfig = blake2b.Config{Size: 20}
+var checksumHashConfig = blake2b.Config{Size: 4}
+var Encoding = base32.NewEncoding(EncodeStd)
+
+// Filecoin address builder
+type AddressBuilder struct {
+	network   string
+	alghoritm xc.SignatureType
+}
+
+var _ xc.AddressBuilder = AddressBuilder{}
+
+// NewAddressBuilder creates a new address builder and validate provided algorithm.
+// Default algorithm is specified in `Driver.SignatureAlgorithm()`
+func NewAddressBuilder(asset xc.ITask, options ...xcaddress.AddressOption) (xc.AddressBuilder, error) {
+	opts, err := xcaddress.NewAddressOptions(options...)
+	if err != nil {
+		return AddressBuilder{}, err
+	}
+
+	algorithm, ok := opts.GetAlgorithmType()
+	// Check if default algorithm is overridden
+	if ok {
+		if algorithm != xc.K256Sha256 {
+			return AddressBuilder{}, fmt.Errorf("unsupported address type: %s", algorithm)
+		}
+	} else {
+		algorithm = asset.GetChain().Driver.SignatureAlgorithm()
+	}
+
+	return AddressBuilder{
+		network:   asset.GetChain().Net,
+		alghoritm: algorithm,
+	}, nil
+}
+
+func (ab AddressBuilder) GetSecp256k1Address(publicKeyBytes []byte) (xc.Address, error) {
+	pk, err := secp.ParsePubKey(publicKeyBytes)
```

Comment:
Can you use btcec dependency instead?  I think it does the same thing.  Will allow us to remove this `secp256k1` dependency.

```go
import "github.com/btcsuite/btcd/btcec/v2"

	pubKey, err := btcec.ParsePubKey(serializedPubKey)
	if err != nil {
		return nil, err
	}
```